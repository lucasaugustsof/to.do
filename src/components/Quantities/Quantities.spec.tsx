import { describe, it, expect } from 'vitest'

import { render, screen } from '@testing-library/react'

import { Quantities } from '.'

describe('Components/Quantities', () => {
  it('should render the Quantities component correctly', () => {
    const { asFragment } = render(<Quantities />)

    expect(asFragment()).toMatchSnapshot()
  })

  it('should return the text 0 of 10 and not 0', async () => {
    const testID = 'test:quantities'

    render(<Quantities testID={testID} created={10} />)

    const counterComponent = screen.getByTestId(testID).children[1].children[1]

    expect(counterComponent.innerHTML).not.toBe('0')
  })
})
