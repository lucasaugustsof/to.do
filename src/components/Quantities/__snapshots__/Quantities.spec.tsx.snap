// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Components/Quantities > should render the Quantities component correctly 1`] = `
<DocumentFragment>
  <div
    class="_container_91d335"
  >
    <div>
      <strong>
        Tarefas criadas
      </strong>
      <span
        class="_counter_91d335"
      >
        0
      </span>
    </div>
    <div>
      <strong>
        Conclu√≠das
      </strong>
      <span
        class="_counter_91d335"
      >
        0
      </span>
    </div>
  </div>
</DocumentFragment>
`;
